hudi-client/hudi-client-common/src/main/java/org/apache/hudi/metadata/HoodieBackedTableMetadataWriter.java

```diff
@@ -212,6 +212,9 @@ private HoodieMetadataFileSystemView getMetadataView() {
     if (metadataView == null || !metadataView.equals(metadata.getMetadataFileSystemView())) {
       ValidationUtils.checkState(metadata != null, "Metadata table not initialized");
       ValidationUtils.checkState(dataMetaClient != null, "Data table meta client not initialized");
+      if (metadataView != null) {
+        metadataView.close();
+      }
```

Comment:
Why are we closing it here? Should it be moved to close() method?

Reply:
we are creating a new metadataView here. So, I am just explicilty closing the previous one